drivers/mdio/mdio_shell.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2021 IP-Logix Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <shell/shell.h>
+#include <stdlib.h>
+#include <drivers/mdio.h>
+#include <string.h>
+#include <sys/util.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(mdio_shell, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define MDIO_DEVICE "MDIO"
+
+/*
+ * Scan the entire 5-bit address space of the MDIO bus
+ *
+ * scan [<dev_addr>]
+ */
+static int cmd_mdio_scan(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int cnt;
+	uint16_t data;
+	uint16_t dev_addr;
+
+	dev = device_get_binding(MDIO_DEVICE);
+	if (!dev) {
+		shell_error(sh, "MDIO: Device driver %s not found.",
+			    MDIO_DEVICE);
+		return -ENODEV;
+	}
+
+	if (argc >= 2) {
+		dev_addr = strtol(argv[1], NULL, 16);
+	} else {
+		dev_addr = 0;
+	}
+
+	shell_print(sh,
+		    "Scanning bus for devices. Reading register 0x%x",
+		    dev_addr);
+	cnt = 0;
+
+	mdio_bus_enable(dev);
+
+	for (int i = 0; i < 32; i++) {
+		data = 0;
+		if (mdio_read(dev, i, dev_addr, &data) >= 0 &&
+			data != 0xffff) {
+			cnt++;
+			shell_print(sh, "Found MDIO device @ 0x%x", i);
+		}
+	}
+
+	mdio_bus_disable(dev);
+
+	shell_print(sh, "%u devices found on %s", cnt, MDIO_DEVICE);
+
+	return 0;
+}
+
+/* mdio write <port_addr> <dev_addr> <data> */
```

Comment:
not help too much, you can drop it.

Reply:
Do you mean to drop the function or remove the comment?

Reply:
comment, you define those variables and passes to mdio_write